mlrun/api/api/endpoints/feature_sets.py

```diff
@@ -137,6 +143,77 @@ def list_feature_sets(
     return feature_sets
 
 
+def _needs_v3io_mount(data_source, data_targets):
+    paths = []
+    for target in data_targets:
+        # If the target does not have a path (i.e. default target), then retrieve the default path from config.
+        paths.append(target.path or get_default_prefix_for_target(target.kind))
+    paths.append(data_source.path)
+
+    return any(
+        path.startswith("v3io://") or path.startswith("v3ios://") for path in paths
+    )
+
+
+@router.post(
+    "/projects/{project}/feature-sets/{name}/references/{reference}/ingest",
+    response_model=schemas.FeatureSetIngestOutput,
+)
+def feature_set_ingest(
+    request: Request,
+    project: str,
+    name: str,
+    reference: str,
+    ingest_parameters: schemas.FeatureSetIngest,
+    username: str = Header(None, alias="x-remote-user"),
+    db_session: Session = Depends(deps.get_db_session),
+):
+    tag, uid = parse_reference(reference)
+    feature_set_record = get_db().get_feature_set(db_session, project, name, tag, uid)
+
+    feature_set = mlrun.feature_store.FeatureSet.from_dict(feature_set_record.dict())
+    # Need to override the default rundb since we're in the server.
+    feature_set._override_run_db(db_session)
+
+    data_source = DataSource.from_dict(ingest_parameters.source.dict())
+    data_targets = [
+        DataTargetBase.from_dict(data_target.dict())
+        for data_target in ingest_parameters.targets
+    ]
+
+    run_config = RunConfig()
+
+    # Try to deduce whether the ingest job will need v3io mount, by analyzing the paths to the source and
+    # targets. If it needs it, apply v3io mount to the run_config. Note that the access-key and username are
+    # user-context parameters, we cannot use the api context.
+    if _needs_v3io_mount(data_source, data_targets):
+        secrets = get_secrets(request)
+        access_key = secrets.get("V3IO_ACCESS_KEY", None)
+
+        if not access_key or not username:
+            log_and_raise(
+                HTTPStatus.BAD_REQUEST.value,
+                reason="Request needs v3io access key and username in header",
+            )
+        run_config = run_config.apply(mount_v3io(access_key=access_key, user=username))
+
+    infer_options = ingest_parameters.infer_options or InferOptions.default()
+
+    run_params = ingest(
+        feature_set,
+        data_source,
+        data_targets,
+        infer_options=infer_options,
+        return_df=False,
+        run_config=run_config,
+    )
+    # ingest may modify the feature-set contents, so returning the updated feature-set.
+    result_feature_set = schemas.FeatureSet(**feature_set.to_dict())
+    return schemas.FeatureSetIngestOutput(
```

Comment:
Actually, only need (1), since (2) and (3) are only needed when not returning the default value specified in the `status_code` parameter. Will do the change as needed.

Reply:
Walla, I didn't know, Cool